src/com/seafile/seadroid2/cameraupload/CameraUploadManager.java

```diff
@@ -1,148 +1,103 @@
-        account = act;
-        dbHelper = CameraUploadDBHelper.getInstance();
-        mDataManager = new DataManager(act);
+    public CameraUploadManager(Context context) {
+        accountManager = new AccountManager(context);
     }
 
     /**
-     * get cached photo from local database
+     * Is camera upload enabled?
      *
-     * @param repoName
-     * @param repoID
-     * @param path
-     * @return
+     * @return true if camera upload is enabled.
      */
-    public SeafCachedPhoto getCachedPhoto(String repoName, String repoID,
-            String path) {
-        SeafCachedPhoto cp = dbHelper.getPhotoCacheItem(repoID, path);
-        return cp;
+    public boolean isCameraUploadEnabled() {
+        Account account = getCameraAccount();
+        return account != null;
     }
 
     /**
-     * get cached photo from local database
+     * Get the account that is currently the remote target for the camera upload
      *
-     * @param repoName
-     * @param repoID
-     * @param dir
-     * @param path
-     * @return
+     * @return the account if camera is enabled, null otherwise.
      */
-    public SeafCachedPhoto getCachedPhoto(String repoName, String repoID,
-            String dir, String path) {
-        String validPath = Utils.pathJoin(dir, path);
-        return getCachedPhoto(repoName, repoID, validPath);
+    public Account getCameraAccount() {
+        for (Account account: accountManager.getAccountList()) {
+            int isSyncable = ContentResolver.getIsSyncable(account.getAndroidAccount(), AUTHORITY);
+            if (isSyncable > 0)
+                return account;
+        }
+        return null;
     }
 
     /**
-     * add a photo cache to local database
-     *
-     * @param repoName
-     * @param repoID
-     * @param path
+     * Initiate a camera sync immediately.
      */
-    public void addCachedPhoto(String repoName, String repoID, String path) {
-        SeafCachedPhoto item = new SeafCachedPhoto();
-        item.repoName = repoName;
-        item.repoID = repoID;
-        item.path = path;
-        item.accountSignature = account.getSignature();
-        dbHelper.savePhotoCacheItem(item);
+    public void performSync() {
+        Account cameraAccount = getCameraAccount();
+        if (cameraAccount != null)
+            ContentResolver.requestSync(cameraAccount.getAndroidAccount(), AUTHORITY, Bundle.EMPTY);
     }
 
     /**
-     * remove a photo cache from local database
-     *
-     * @param cp
-     * @return 1 when successfully removed a cache, otherwise 0
-     *
+     * Initiate a camera sync immediately, upload all media files again.
      */
-    public int removeCachedPhoto(SeafCachedPhoto cp) {
-        // unused method
-        // check if the file deletion succeeds
-        return dbHelper.removePhotoCacheItem(cp);
-    }
+    public void performFullSync() {
+        Bundle b = new Bundle();
+        b.putBoolean(ContentResolver.SYNC_EXTRAS_INITIALIZE, true);
```

Comment:
Ok, now I understand your meaning, we use ContentResolver.SYNC_EXTRAS_INITIALIZE in `onPerformSync` to decide if delete the data table and do a full sync. That makes sense.